rbac/management/management/commands/sync_schemas.py

```diff
@@ -0,0 +1,148 @@
+#
+# Copyright 2021 Red Hat, Inc.
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""Schema sync command."""
+
+from django.core.management.base import BaseCommand
+from django.db.utils import IntegrityError
+from management.models import Group, Policy, Principal, Role
+from tenant_schemas.utils import tenant_context
+
+from api.models import Tenant
+
+
+class Command(BaseCommand):
+    """Command for use with django's manage.py script."""
+
+    help = "Syncs custom principals, roles, groups, and policies to the public schema"
+
+    def clear_pk(self, entry):
+        """Clear the ID and PK values for provided postgres entry."""
+        entry.id = None
+        entry.pk = None
+
+    def copy_custom_principals_to_public(self, tenant):
+        """Copy custom principals from provided tenant to the public schema."""
+        principals = Principal.objects.all()
+        public_schema = Tenant.objects.get(schema_name="public")
+        for principal in principals:
+            self.stdout.write(f"Copying principal {principal.username} to public schema for tenant {tenant}.")
+            if not principal.tenant:
+                principal.tenant = tenant
+                principal.save()
+            with tenant_context(public_schema):
+                self.clear_pk(principal)
+                try:
+                    principal.save()
+                except IntegrityError as err:
```

Comment:
We've got a lot of error handling and formatting code throughout the codebase. It might be worth working on moving some of that here and letting this custom handler take care of that for us?